### PR TITLE
feat: add make_regional_gl_entries override for Sales Invoice

### PR DESCRIPTION
**Pull request by FHenry**
> Explain the **details** for making this change. What existing problem does the pull request solve?

following review on #42278

For a common feature in France/Europe (but missing in ERPNext core), we've introduce in France Regional Localization app (https://github.com/scopen-coop/erpnext_france), the "Deposit invoice". That's a real Sales invoices / Purchase Invoice issuing (not Payment Request) based on % or fixed amount from a Sales Order / Purchase Order .
 
We need to adapt make_gl_entries and make_reverse_gl_entries method because the actual tax accounting is not correct according regional accountancy law.
Income amount from this type of invoice can have a different Account, and Tax amount could be accounted in split/spread Accounts if it talk about Goods or Service or also, but not only, depend Customer/Supplier country, and other case...

Actually, we almost manage to update GL entry after core ERPNext make_gl_entries/make_reverse_gl_entries but that not the good way of doing things.
This missing "@erpnext.allow_regional" on these methods force us to spend a lot's of time to "hack" ERPNext, and in some case do not allow to implement the legal requirement for France accounting law.

Let me know what do you think about this PR

> no-docs

### CODE REVIEW

File: erpnext/controllers/stock_controller.py [hunk 1]
```diff
@@ -581,6 +581,8 @@ def get_gl_entries(self, warehouse_account=None, default_expense_account=None, d
 						).format(wh, self.company)
 					)
 
+		gl_list = make_regional_gl_entries(gl_list, self)
```
Comment (vorasmit) [inline erpnext/controllers/stock_controller.py]: What kind of changes are you looking for in your custom app (for stock)? Which doctype is it?

`get_gl_entries` is internally used by `make_gl_entries` and overriding one should be enough in my opinion.
So if you wish to update stock gl entries for sales invoice, you can do so with regional override created above.

Comment (FHenry) [reply erpnext/controllers/stock_controller.py]: I was thinking about special use case with Materiel Renting  where stock valuation need to be adapted.
But, Scopen only had a prospect now with this feature, let's remove it the form this PR, to keep only on Sales Invoices regional feature., I've updated the PR title

Comment (vorasmit) [reply erpnext/controllers/stock_controller.py]: [This](https://github.com/frappe/erpnext/blob/develop/erpnext/controllers/buying_controller.py#L336) hook could be useful for all purchases.

Can be done similarly for Sales if needed.

**Comment by FHenry** [comment]
@vorasmit 
Could you add backport version 15 label ?

**Pull request merged**